models/pcs/pcs.h

```diff
@@ -0,0 +1,115 @@
+#define NIGHT_FACTOR          0.2
+
+// LVT is in seconds, so we need to convert it
+#define HOUR            3600
+#define DAY             (24 * HOUR)
+#define WEEK            (7 * DAY)
+
+// Definition of the time phases within each day
+#define EARLY_MORNING   (6.5 * HOUR)
+#define MORNING         (13 * HOUR)
+#define LUNCH           (15 * HOUR)
+#define AFTERNOON       (19 * HOUR)
+#define EVENING         (21 * HOUR)
+
+// Events
+#define START_CALL	20
+#define END_CALL	21
+#define HANDOFF_LEAVE	30
+#define HANDOFF_RECV	31
+
+// Macros to manage channel availability as a bitmap
+#define MSK 1ULL
+#define BITS (sizeof(unsigned long long) * CHAR_BIT)
+#define CHECK_CHANNEL(P,I) ( (P)[(I) / BITS] & (MSK << ((I) % BITS)) )
+#define SET_CHANNEL(P,I) ( (P)[(I) / BITS] |= (MSK << ((I) % BITS)) )
+#define RESET_CHANNEL(P,I) ( (P)[(I) / BITS] &= ~(MSK << ((I) % BITS)) )
+
+
+// Accurate simulation based on the model in:
+// S. Kandukuri and S. Boyd,
+// "Optimal power control in interference-limited fading wireless channels with outage-probability specifications,"
+// in IEEE Transactions on Wireless Communications, vol. 1, no. 1, pp. 46-55, Jan. 2002, doi: 10.1109/7693.975444.
+#define CROSS_PATH_GAIN		0.00000000000005
+#define PATH_GAIN		0.0000000001
+#define MIN_POWER		3
+#define MAX_POWER		3000
+#define SIR_AIM			10
+
+
+enum reason { START, HANDOFF };
+
+// Event Structure
+struct event_content {
+	unsigned cell;
+	int channel;
+	double end_time;
+	enum reason r;
+};
+
+// Structs to define the simulation state
+struct channel {
+	int channel_id;
+	double fading;
+	double power;
+	struct channel *next;
+};
+
+struct lp_state {
+	double lvt;
+	double ta;
+
+	unsigned int channel_counter;
+	unsigned int arriving_calls;
+	unsigned int complete_calls;
+	unsigned int blocked_on_setup;
+	unsigned int blocked_on_handoff;
+	unsigned int handoff_leaving;
+	unsigned int handoff_entering;
+
+	unsigned long long *channel_state;
+	struct channel channels;
+};
+
+
+/*
+typedef struct _lp_state_type{
+	unsigned int channel_counter; // How many channels are currently free
+	unsigned int arriving_calls; // How many calls have been delivered within this cell
+	unsigned int complete_calls; // Number of calls which were completed within this cell
+	unsigned int blocked_on_setup; // Number of calls blocked due to lack of free channels
+	unsigned int blocked_on_handoff; // Number of calls blocked due to lack of free channels in HANDOFF_RECV
+	unsigned int leaving_handoffs; // How many calls were diverted to a different cell
+	unsigned int arriving_handoffs; // How many calls were received from other cells
+	unsigned int cont_no_sir_aim; // Used for fading recheck
+	unsigned int executed_events; // Total number of events
+
+	simtime_t lvt; // Last executed event was at this simulation time
+
+	double ta; // Current call interarrival frequency for this cell
+
+	unsigned int *channel_state;
+	struct _channel *channels;
+} lp_state_type;
+*/
```

Comment:
I'm not sure we need this block of commented code :monocle_face:

Reply:
No we don't! :+1: